sklearn/utils/estimator_checks.py

```diff
@@ -807,26 +825,30 @@ def check_fit_score_takes_y(name, Estimator):
         if func is not None:
             func(X, y)
             args = [p.name for p in signature(func).parameters.values()]
+            if args[0] == "self":
+                # if_delegate_has_method makes methods into functions
+                # with an explicit "self", so need to shift arguments
+                args = args[1:]
```

Comment:
Wow, that was pretty subtle! For reference I made [this gist](https://gist.github.com/vene/def1a57b5c233bde285d9da5ac616ba0).